src/lib.rs

```diff
@@ -344,11 +344,18 @@ where
         Pool::new_inner(config, manager, Duration::from_secs(30))
     }
 
-    // for testing
-    fn new_inner(
+    /// Create a new connection pool with a shared thread_pool
+    ///
+    /// The `config.helper_threads` will be ignored in this situation.
+    ///
+    /// Returns an `Err` value if `initialization_fail_fast` is set to true in
+    /// the configuration and the pool is unable to open all of its
+    /// connections.
+    pub fn with_thread_pool(
         config: Config<M::Connection, M::Error>,
         manager: M,
         reaper_rate: Duration,
+        thread_pool: Arc<ScheduledThreadPool>,
```

Comment:
Seems like it might make more sense to add this to `Config` rather than adding a new constructor method?

Reply:
It's true, updated